x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -240,56 +247,77 @@ const GcpInputVarFields = ({
+  const credentialsTypeFields = getFieldById('credentials_type');
+  const credentialFilesFields = getFieldById('credentials_file');
+  const credentialJSONFields = getFieldById('credentials_json');
+  const credentialsTypeField = {
+    id: 'credentials_type',
+    label: 'Credentials Type',
+    type: 'text',
+    value: credentialsTypeFields || credentialOptionsList[0].value,
+  };
+  const credentialFieldValue = credentialOptionsList[0].value;
+  const credentialJSONValue = credentialOptionsList[1].value;
+
   return (
     <div>
       <EuiForm component="form">
-        <EuiFormRow fullWidth label={gcpField.fields.project_id.label}>
-          <EuiFieldText
-            id={targetFieldName('project_id')!.id}
-            fullWidth
-            value={targetFieldName('project_id')!.value || ''}
-            onChange={(event) => onChange(targetFieldName('project_id')!.id, event.target.value)}
-          />
-        </EuiFormRow>
-        <EuiFormRow fullWidth label={'Credentials'}>
-          <EuiSelect
-            fullWidth
-            options={credentialOptionsList}
-            value={credentialOption}
-            onChange={(optionElem) => {
-              setCredentialOption(optionElem.target.value);
-            }}
-          />
-        </EuiFormRow>
-        {credentialOption === 'Credentials File' && (
-          <EuiFormRow fullWidth label={CredentialFileText}>
+        {projectIdFields && (
+          <EuiFormRow fullWidth label={gcpField.fields.project_id.label}>
             <EuiFieldText
-              id={targetFieldName('credentials_file')!.id}
+              data-test-subj={CIS_GCP_INPUT_FIELDS_TEST_SUBJECTS.PROJECT_ID}
+              id={projectIdFields.id}
               fullWidth
-              value={targetFieldName('credentials_file')!.value || ''}
-              onChange={(event) =>
-                onChange(targetFieldName('credentials_file')!.id, event.target.value)
-              }
+              value={projectIdFields.value || ''}
+              onChange={(event) => onChange(projectIdFields.id, event.target.value)}
             />
           </EuiFormRow>
         )}
-        {credentialOption === 'Credentials JSON' && (
-          <EuiFormRow fullWidth label={CredentialJSONText}>
+        {credentialFilesFields && credentialJSONFields && (
+          <EuiFormRow fullWidth label={gcpField.fields.credentials_type.label}>
+            <EuiSelect
+              data-test-subj={CIS_GCP_INPUT_FIELDS_TEST_SUBJECTS.CREDENTIALS_TYPE}
+              fullWidth
+              options={credentialOptionsList}
+              value={credentialsTypeFields?.value}
+              onChange={(optionElem) => {
+                onChange(credentialsTypeField.id, optionElem.target.value);
+              }}
+            />
+          </EuiFormRow>
+        )}
+
+        {(credentialsTypeFields?.value || credentialsTypeField.value) === credentialFieldValue &&
+          credentialFilesFields && (
```

Comment:
after updating from my previous update we can use only credentialsTypeField for checking

```jsx
{credentialsTypeFields.value === credentialFieldValue && credentialFilesFields && (
```